File: merge-products.js
const fs = require('fs');
const path = require('path');

const currentProductsPath = path.join(__dirname, 'data', 'products.json');
const oldProductsPath = path.join(__dirname, 'old-products.json');

const currentProducts = JSON.parse(fs.readFileSync(currentProductsPath, 'utf8'));
const oldProducts = JSON.parse(fs.readFileSync(oldProductsPath, 'utf8'));

const productMap = new Map(currentProducts.map(p => [p.product_id, p]));

oldProducts.forEach(oldProduct => {
    if (!productMap.has(oldProduct.product_id)) {
        productMap.set(oldProduct.product_id, oldProduct);
    }
});

const mergedProducts = Array.from(productMap.values());

fs.writeFileSync(currentProductsPath, JSON.stringify(mergedProducts, null, 2));

console.log(`Merge complete. Total products: ${mergedProducts.length}`); 

File: phone-change-backup-20251006-154439/production stuff/merge_products.py
import json

# Load the new products (from staging)
with open('production stuff/data/products.json', 'r') as f:
    new_products = json.load(f)

# Load the old products (complete list)
with open('old-products.json', 'r') as f:
    old_products = json.load(f)

# Create a dictionary of the new products for easy lookup
new_products_dict = {p['product_id']: p for p in new_products}

# Create the final merged list
merged_products = []

# Iterate through the old products, and update with new data where available
for old_product in old_products:
    if old_product['product_id'] in new_products_dict:
        # This product exists in the new data, so use the new data
        merged_products.append(new_products_dict[old_product['product_id']])
    else:
        # This product only exists in the old data, so keep it
        merged_products.append(old_product)

# Write the final merged list to the products.json file
with open('production stuff/data/products.json', 'w') as f:
    json.dump(merged_products, f, indent=2)

print(f"Merge complete. Total products: {len(merged_products)}")
